regex.c

```diff
@@ -38,6 +39,14 @@ static bool init_regex(const char *regexin, int index)
 		log_regex_error("compiling", errcode, index);
 		return false;
 	}
+
+	// Store compiled regex string in buffer if in regex debug mode
+	if(config.regex_debugmode)
+	{
+		regexbuffer[index] = strdup(regexin);
+		if(regexbuffer[index] == NULL)
+			logg("WARN: init_regex() failed to allocate memory for regexbuffer[index]");
```

Comment:
We already have these allocation and free cases handled in `memory.c`, and they provide more detail.